os/arch/arm/src/armv7-m/up_doirq.c

```diff
@@ -76,6 +76,7 @@
 #ifdef CONFIG_ARCH_NESTED_INTERRUPT
 extern uint32_t g_nestlevel; /* Initial top of interrupt stack */
 #endif
+int g_irq_nums[2] = {0}; /* Array to store the last two interrupt numbers */
```

Comment:
1. Does this save two interrupts? or an interrupt? If it's 2, why do you check only index 1?
```
	int assertInIrq = g_irq_nums[1];
	if (assertInIrq) {
```
2. Do you use g_irq_num[0] at anywhere? If not, why do you save?

Reply:
1) Yes because while handling the interrupt, the value of the r0 is overwritten and we need this overwritten value to track the interrupt in which the crash has occurred.
2) This is used to track the irq value which is later copied to the g_irq_num[1]. Henc eit is required.